BAZProject/MainMovie/MainMovieRouter.swift

```diff
@@ -10,21 +10,47 @@
 
 import UIKit
 
-class MainMovieRouter: MainMovieWireframeProtocol {
+class MainMovieRouterFactory {
```

Comment:
Recuerda marcar la clase como final 
```suggestion
final class MainMovieRouterFactory {
```

También creo que el nombre debería cambiar un poco. El nombre actual `MainMovieRouterFactory` me da a entender que este factory construirá un MainMovieRouter, cuando en realidad estás construyendo un `MainMovieViewController`. Tal vez el nombre `MainMovieViewControllerFactory` sería más adecuado.